spec/lib/finder_api_spec.rb

```diff
@@ -20,29 +20,72 @@ def result_item(id, title, score:)
             {
               "results" => [
-                result_item("/register-to-vote", "Register to Vote", score: 1),
+                result_item("/register-to-vote", "Register to Vote", score: 1, updated: "14-12-19", popularity: 3),
               ],
             },
             {
               "results" => [
-                result_item("/hmrc", "HMRC", score: 10),
-                result_item("/register-to-vote", "Register to Vote", score: 2),
+                result_item("/hmrc", "HMRC", score: 10, updated: "14-12-18", popularity: 2),
+                result_item("/register-to-vote", "Register to Vote", score: 2, updated: "14-12-19", popularity: 3),
               ],
             },
           ]
-        )
+      )
+    end
+
+    context 'default' do
+      subject { described_class.new("/finder", {}).content_item_with_search_results }
+
+      it "de-duplicates and sorts by public_updated decending" do
+        results = subject.fetch("details").fetch("results")
+        expect(results.first).to match(hash_including("_id" => "/register-to-vote"))
+        expect(results.second).to match(hash_including("_id" => "/hmrc"))
+      end
+    end
+
+    context 'updated-newest' do
+      subject { described_class.new("/finder", {}, order: 'updated-newest').content_item_with_search_results }
+
+      it "de-duplicates and sorts by public_updated decending" do
+        results = subject.fetch("details").fetch("results")
+        expect(results.first).to match(hash_including("_id" => "/register-to-vote"))
+        expect(results.second).to match(hash_including("_id" => "/hmrc"))
+      end
+    end
+
+    context 'updated-oldest' do
+      subject { described_class.new("/finder", {}, order: 'updated-oldest').content_item_with_search_results }
+
+      it "de-duplicates and sorts by public_updated decending" do
+        results = subject.fetch("details").fetch("results")
+        expect(results.first).to match(hash_including("_id" => "/hmrc"))
+        expect(results.second).to match(hash_including("_id" => "/register-to-vote"))
+      end
     end
 
-    it "de-duplicates the content" do
-      results = subject.fetch("details").fetch("results")
-      expect(results.first).to match(hash_including("_id" => "/register-to-vote"))
-      expect(results.second).to match(hash_including("_id" => "/hmrc"))
+    context 'most-viewed' do
+      subject { described_class.new("/finder", {}, order: 'most-viewed').content_item_with_search_results }
+
+      it "de-duplicates and sorts by popularity descending" do
+        results = subject.fetch("details").fetch("results")
+        expect(results.first).to match(hash_including("_id" => "/register-to-vote"))
+        expect(results.second).to match(hash_including("_id" => "/hmrc"))
+      end
+    end
+
+    context 'a-z' do
+      subject { described_class.new("/finder", {}, order: 'a-z').content_item_with_search_results }
+
+      it "de-duplicates and sorts by title descending" do
```

Comment:
Is there a `z-a` sort option as well?

Reply:
There isn't at the moment. Only 4 sort options specified at the moment.
* Most relevant
* Most viewed
* Most recent
* A to Z